dashboard/app/controllers/followers_controller.rb

```diff
@@ -64,6 +72,10 @@ def load_section
     # Note that we treat the section as not being found if the section user
     # (i.e., the teacher) does not exist (possibly soft-deleted) or is not a teacher
     unless @section && @section.user&.teacher?
+      # Adjust user properties to track failed section join attempts
+      if current_user
```

Comment:
Hmmm.  Is there a loophole here where a non-signed-in user can keep trying to join a section?

Reply:
Yes, this has been sitting in the back of my mind for a bit. If you hit `/join/:section_code` while not signed in, you need to fill out a form to create an account after which you will automatically join the section indicates by the section_code parameter in the route. We can't track this behavior with the current implementation because a new user would always start with zero section attempts, thus not requiring captcha validation. In my opinion, the only way to fully solve this issue is to stick another captcha at the bottom of the form where the user creates an account. Not sure we have the ability to implement any fancy logic if we go that route since the user won't yet exist; thus we can't store anything server-side.

Reply:
I think what you see after joining depends on whether the accounts are student-owned or teacher-owned?  In my test class where I use word logins, I get the "select a student" page when joining from a logged out page.

I'm wondering if a quick and reasonable mitigation here might be to always show the captcha if the user is not logged in.  (Feel free to follow-up on this separately from this PR if you prefer.)